src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -878,7 +878,30 @@ private static bool IsGlobalAttributeTarget(SyntaxToken token)
 
         private bool IsPossibleAttributeDeclaration()
         {
-            return this.CurrentToken.Kind == SyntaxKind.OpenBracketToken;
+            // Have to at least start with `[` to be an attribute
+            if (this.CurrentToken.Kind != SyntaxKind.OpenBracketToken)
+                return false;
+
+            using var _ = this.GetDisposableResetPoint(resetOnDispose: true);
+
+            // Eat the `[`
+            EatToken();
+
+            // `[ id` is always an attribute.
```

Comment:
>is always an attribute. [](http://example.com/codeflow?start=22&length=23)

Perhaps not "always", but "very likely"?

Reply:
updated.